### PR TITLE
Sparse logistic regression

### PR DESCRIPTION
**Pull request by henryre**
* Adds `SparseLogisticRegression` which gives at least a 10x speed up for typical Snorkel feature matrices over `LogisticRegression`
* Updates tutorials
* Small fix to GridSearch
* Adds nnz for weight vector

### CODE REVIEW

**Comment by henryre** [comment]
Travis is whack. Just reran from a fresh clone and everything ran correctly.

File: tutorials/intro/Intro_Tutorial_2.ipynb [hunk 1]
```diff
@@ -16,7 +16,7 @@
   },
```
Comment (ajratner) [inline tutorials/intro/Intro_Tutorial_2.ipynb]: Just clear and refresh this notebook for easier merge

File: snorkel/learning/logistic_regression.py [hunk 2]
```diff
@@ -109,3 +117,90 @@ def save_info(self, model_name):
...
+
+    def _batch_sparse_data(self, X):
+        """Convert sparse batch matrix to sparse inputs for embedding lookup"""
+        if not issparse(X):
+            raise Exception("Matrix X must be scipy.sparse type")
+        X_lil = X.tolil()
+        indices, ids, weights = [], [], []
+        max_len = 0
+        for i, (row, data) in enumerate(zip(X_lil.rows, X_lil.data)):
+            max_len = max(max_len, len(row))
```
Comment (ajratner) [inline snorkel/learning/logistic_regression.py]: Why would this be < 0?  Maybe comment here to clarify?

File: snorkel/learning/logistic_regression.py [hunk 3]
```diff
@@ -109,3 +117,90 @@ def save_info(self, model_name):
...
+            indices.extend((i, t) for t in xrange(len(row)))
+            ids.extend(row)
+            weights.extend(data)
+        shape = (len(X_lil.rows), max_len)
+        return indices, shape, ids, weights
+
+    def _run_batch(self, X_train, y_train, i, r):
+        """Run a single batch update"""
+        # Get batch sparse tensor data
+        indices, shape, ids, weights = self._batch_sparse_data(X_train[i:r, :])
```
Comment (ajratner) [inline snorkel/learning/logistic_regression.py]: Is it worth converting `X_train` to CSR first before doing these cuts for each batch?

File: snorkel/learning/logistic_regression.py (hunk 2 above)
Comment (henryre) [inline snorkel/learning/logistic_regression.py]: It wouldn't. I'm just looking for the max over all the rows.

**Pull request merged**